diff --git a/include/pybind11/embed.h b/include/pybind11/embed.h
index 683f2428..674d5fec 100644
--- a/include/pybind11/embed.h
+++ b/include/pybind11/embed.h
@@ -15,9 +15,10 @@
 #include <memory>
 #include <vector>
 
-#if defined(PYPY_VERSION)
-#    error Embedding the interpreter is not supported with PyPy
-#endif
+// MIPT: enable embedding to succeed build on PyPy.
+// #if defined(PYPY_VERSION)
+// #    error Embedding the interpreter is not supported with PyPy
+// #endif
 
 #if PY_MAJOR_VERSION >= 3
 #    define PYBIND11_EMBEDDED_MODULE_IMPL(name)                                                   \
@@ -96,7 +97,9 @@ struct wide_char_arg_deleter {
 };
 
 inline wchar_t *widen_chars(const char *safe_arg) {
-#if PY_VERSION_HEX >= 0x030500f0
+// MIPT: PyPy does not support Py_DecodeLocale C API and is complex to implement.
+//       Use internal logic.
+#if PY_VERSION_HEX >= 0x030500f0 && !defined(PYPY_VERSION)
     wchar_t *widened_arg = Py_DecodeLocale(safe_arg, nullptr);
 #else
     wchar_t *widened_arg = nullptr;
